<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Console launch prefix -->
  <arg name="output"        default="screen"/>

  <!-- Config and weights folder. -->
  <arg name="aligned_reid_model"         default="$(find jpda_rospack)/src/aligned_reid_JRDB_weights.pth"/>
  <arg name="fpointnet_config"           default="$(find jpda_rospack)/config/featurepointnet.cfg"/>
  <arg name="calib_3d"                   default="$(find jpda_rospack)/calib"/>
  <arg name="combination_depth_weight"   default="1"/>
  <arg name="combination_model_path"     default="0"/>
  <arg name="sequence_name"   default="bytes-cafe-2019-02-07_0"/>

<rosparam>use_sim_time: true</rosparam>

<node pkg="image_transport" type="republish" name="camera9" output="log" 
    args="compressed in:=/ros_indigosdk_node/stitched_image0 
          raw out:=/ros_indigosdk_node/stitched_image0"/>

  <node pkg="jpda_rospack" type="store_features.py" name="store_features_node" output="$(arg output)" respawn="false">
  <param name="sequence_name"       value="$(arg sequence_name)" />
  </node>

 <node pkg="jpda_rospack" type="template.py" name="jpda_aligned_reid" output="screen" respawn="false">
    <param name="aligned_reid_model"          value="$(arg aligned_reid_model)" />
  </node> 

  <node pkg="jpda_rospack" type="3d_detector.py" name="jpda_3d_detector" output="screen" respawn="false">
    <param name="fpointnet_config"          value="$(arg fpointnet_config)" />
    <param name="calib_3d"                  value="$(arg calib_3d)" />
  </node>
  
  <!--<node pkg="jpda_rospack" type="tracker_3d_node.py" name="jpda_tracker_3d" output="$(arg output)" respawn="false">
    <param name="combination_depth_weight"           value="$(arg combination_depth_weight)" />
    <param name="calib_3d"                           value="$(arg calib_3d)" />
    <param name="combination_model_path"             value="$(arg combination_model_path)" />
  </node>-->

</launch>
